lib/iris/cube.py

```diff
@@ -1738,8 +1737,7 @@ def data(self):
                 result = as_concrete_data(self._dask_array,
                                           nans_replacement=ma.masked,
                                           result_dtype=self.dtype)
-                self._numpy_array = result
-                self.dtype = None
+                self.data = result
```

Comment:
Changing the instance's `data` attribute via the data setter ensures all the appropriate processing for the new data value is applied when the instance's lazy data is computed.